src/query/api/v1/handler/fetch_options.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package handler
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/m3db/m3/src/query/storage"
+	"github.com/m3db/m3/src/x/net/http"
+)
+
+// FetchOptionsBuilder builds fetch options based on a request and default
+// config.
+type FetchOptionsBuilder interface {
+	NewFetchOptions(req *http.Request) (*storage.FetchOptions, *xhttp.ParseError)
+}
+
+// FetchOptionsBuilderOptions provides options to use when creating a
+// fetch options builder.
+type FetchOptionsBuilderOptions struct {
+	Limit int
+}
+
+type fetchOptionsBuilder struct {
+	opts FetchOptionsBuilderOptions
+}
+
+// NewFetchOptionsBuilder returns a new fetch options builder.
+func NewFetchOptionsBuilder(
+	opts FetchOptionsBuilderOptions,
+) FetchOptionsBuilder {
+	return fetchOptionsBuilder{opts: opts}
+}
+
+func (b fetchOptionsBuilder) NewFetchOptions(
+	req *http.Request,
+) (*storage.FetchOptions, *xhttp.ParseError) {
+	fetchOpts := storage.NewFetchOptions()
+	fetchOpts.Limit = b.opts.Limit
+	if str := req.Header.Get(LimitMaxTimeseriesHeader); str != "" {
```

Comment:
Do we have any documentation right now on setting query limits and such? If not would be nice to add a simple document that outlines the YAML configs you can set as well as this head.

Reply:
Yeah, let me look around for it.  @benraskin92 do you know of any?